### PR TITLE
Various additions

### PR DESCRIPTION
**Pull request by YuRaNnNzZZ**
Changes:
- Added Russian localization
- Fixed camera FOV and sensitivity
- Added new optional harder IC2 recipes that uses new Highly Advanced Machine Block
- Added IC2 Basalt and AE2 Sky Stone to space worldgen
- Added two new methods: WarpDriveConfig.getModItem(String mod, String id, int damage) and WarpDriveConfig.getModBlock(String mod, String id) (because locking method for only one mod - bad)
- Added remapping from old ids to new for existing worlds

### CODE REVIEW

File: src/main/java/cr0s/warpdrive/WarpDrive.java [hunk 1]
```diff
@@ -711,6 +727,165 @@ private static void initIC2Recipes() {
...
+		ItemStack energiumDust = WarpDriveConfig.getModItem("IC2", "itemDust2", 2).copy();
+		ItemStack crystalmemory = WarpDriveConfig.getModItem("IC2", "itemcrystalmemory", -1).copy();
+		ItemStack itemHAMachine = new ItemStack(blockHiAdvMachine).copy();
+		
+		GameRegistry.addRecipe(new ItemStack(blockShipCore),"uau", "tmt", "uau",
+				'a', advancedAlloy,
+				't', WarpDriveConfig.getModItem("IC2", "blockMachine2", 0), // Teleporter
+				'm', itemHAMachine,
+				'u', mfsu);
+		if (Loader.isModLoaded("OpenComputers")) {
```
Comment (TheNumenorean) [inline src/main/java/cr0s/warpdrive/WarpDrive.java]: This should use WarpDriveConfig.isOCLoaded

File: src/main/java/cr0s/warpdrive/WarpDrive.java [hunk 2]
```diff
@@ -711,6 +727,165 @@ private static void initIC2Recipes() {
...
+			GameRegistry.addRecipe(new ShapedOreRecipe(new ItemStack(blockShipController), false, new Object[] { "aha", "cmc", "apa", // With OC Adapter
+					'a', advancedAlloy,
+					'm', itemHAMachine,
+					'c', "circuitAdvanced",
+					'h', crystalmemory,
+					'p', WarpDriveConfig.getModItem("OpenComputers", "adapter", -1)}));
+		} else if (Loader.isModLoaded("ComputerCraft")) {
```
Comment (TheNumenorean) [inline src/main/java/cr0s/warpdrive/WarpDrive.java]: Same as above

File: src/main/java/cr0s/warpdrive/conf/WarpDriveConfig.java [hunk 3]
```diff
@@ -759,13 +793,21 @@ private static void loadAdvancedRepulsionSystems() {
 
 	public static Block getDefaultSurfaceBlock(Random random, boolean corrupted, boolean isMoon) {
 		if (isMoon) {
-			if (random.nextInt(5) == 1) {
+			if (isIndustrialCraft2loaded && random.nextInt(10) == 1)
```
Comment (TheNumenorean) [inline src/main/java/cr0s/warpdrive/conf/WarpDriveConfig.java:798]: These world gen changes will be deleted along with this form of world gen soon, but they don't hurt anything for the present

**Comment by TheNumenorean** [comment]
I have mentioned my only issues for the present. Pretty small problems, we will probably merge this in soon.

File: src/main/java/cr0s/warpdrive/WarpDrive.java [hunk 4]
```diff
@@ -197,19 +203,19 @@ public void preInit(FMLPreInitializationEvent event) {
...
-		}
+		// Not needed as we register this at right click on the monitor.
+		/* if (FMLCommonHandler.instance().getSide().isClient()) {
+		 * 	Minecraft mc = Minecraft.getMinecraft();
+		 * 
+		 * 	normalFOV = mc.gameSettings.fovSetting;
+		 * 	normalSensitivity = mc.gameSettings.mouseSensitivity;
+		 * 	logger.info("FOV is " + normalFOV + " Sensitivity is " + normalSensitivity);
+		 * }
+		 */
```
Comment (LemADEC) [inline src/main/java/cr0s/warpdrive/WarpDrive.java]: This code is required to recover from crashes and server kick while being in the monitor.

File: src/main/java/cr0s/warpdrive/WarpDrive.java [hunk 5]
```diff
@@ -795,4 +970,132 @@ public static void addChatMessage(final EntityPlayer player, final String messag
...
+				} else if (mapping.name.equals("WarpDrive:transportBeacon")) {
+					mapping.remap(blockTransportBeacon);
+				} else if (mapping.name.equals("WarpDrive:transporter")) {
+					mapping.remap(blockTransporter);
+				} else if (mapping.name.equals("WarpDrive:warpCore")) {
+					mapping.remap(blockShipCore);
+				}
+			}
+		}
+	}
```
Comment (LemADEC) [inline src/main/java/cr0s/warpdrive/WarpDrive.java:1100]: Thanks! I was about to that tonight :)

File: src/main/java/cr0s/warpdrive/WarpDrive.java (hunk 4 above)
Comment (YuRaNnNzZZ) [inline src/main/java/cr0s/warpdrive/WarpDrive.java]: How code that runs only on client can crash server?

Comment (TheNumenorean) [inline src/main/java/cr0s/warpdrive/WarpDrive.java]: If the player is kicked from the server while in the camera, this fixes it

Comment (TheNumenorean) [inline src/main/java/cr0s/warpdrive/WarpDrive.java]: It's not about crashing the server

Comment (YuRaNnNzZZ) [inline src/main/java/cr0s/warpdrive/WarpDrive.java]: I'm using port of cr0s' version of the mod and monitor is not kicking me from server. I didn't tested this version yet, so i can't tell you that it kicking or not.

Comment (TheNumenorean) [inline src/main/java/cr0s/warpdrive/WarpDrive.java]: For now, can you just remove this change? I would like to take a look at it myself done it is somewhat contested, so if you remove it from this pull request I'll merge the rest right now, then we can take a look at this part again

Comment (TheNumenorean) [inline src/main/java/cr0s/warpdrive/WarpDrive.java]: Even better, Lem would take a look and test the change, he knows more about it than i

**Comment by TheNumenorean** [comment]
Thanks for humoring us

**Pull request merged**